core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -62,21 +63,85 @@ protected DateHistogramAggregationBuilder createFactory(String aggregationName,
         if (extendedBounds != null) {
             factory.extendedBounds(extendedBounds);
         }
-        Boolean keyed = (Boolean) otherOptions.get(HistogramAggregator.KEYED_FIELD);
+        Boolean keyed = (Boolean) otherOptions.get(Histogram.KEYED_FIELD);
         if (keyed != null) {
             factory.keyed(keyed);
         }
-        Long minDocCount = (Long) otherOptions.get(HistogramAggregator.MIN_DOC_COUNT_FIELD);
+        Long minDocCount = (Long) otherOptions.get(Histogram.MIN_DOC_COUNT_FIELD);
         if (minDocCount != null) {
             factory.minDocCount(minDocCount);
         }
-        InternalOrder order = (InternalOrder) otherOptions.get(HistogramAggregator.ORDER_FIELD);
+        InternalOrder order = (InternalOrder) otherOptions.get(Histogram.ORDER_FIELD);
         if (order != null) {
             factory.order(order);
         }
         return factory;
     }
 
+    @Override
+    protected boolean token(String aggregationName, String currentFieldName, Token token, XContentParser parser,
```

Comment:
agreed, I wanted to avoid creating potential issues by changing serialization logic (this code was just moved) but I agree we should move to ObjectParser